generator/standalone_main.cc

```diff
@@ -52,6 +53,69 @@ GetConfig(std::string const& filepath) {
   return google::cloud::Status(google::cloud::StatusCode::kInvalidArgument,
                                "Unable to parse textproto file.");
 }
+
+std::string Basename(std::string const& path) {
+  auto const l = path.find_last_of('/');
+  if (l == std::string::npos) return path;
+  return path.substr(l + 1);
+}
+
+std::string Dirname(std::string const& path) {
+  auto const l = path.find_last_of('/');
+  if (l == std::string::npos) return path;
+  return path.substr(0, l);
+}
```

Comment:
Nit: These may not matter in the particular use cases, but `Dirname("/foo") == ""` and `Dirname("foo") == "foo"` seem wrong without some note about the supported shape of `path`.

Reply:
Done.

Reply:
> The return value for ... paths without `/` is unspecified

Perhaps you should drop the test for `npos` then.  The above is equivalent to `return path.substr(0, path.find_last_of('/'));`.

Reply:
Sure.